contracts/ExitHandler.sol

```diff
@@ -239,18 +239,24 @@ contract ExitHandler is DepositHandler {
         } else if (isNST(currentExit.color)) {
           bytes32 tokenData = exitsTokenData[utxoId];
           address tokenAddr = address(tokens[currentExit.color].addr);
-          IERC1948 nst = IERC1948(tokenAddr);
 
-          // if this fails, we do not care
-          nst.writeData(currentExit.amount, tokenData);
-          tokens[currentExit.color].addr.transferFrom(address(this), currentExit.owner, currentExit.amount);
+          bool success;
+          (success, ) = tokenAddr.call(abi.encodeWithSignature("writeData(uint256,bytes32)", currentExit.amount, tokenData));
+          // if set data did not work, we assume the token hasn't been minted yet
+          if (!success) {
```

Comment:
Wait a moment, don't we lock the contract up in the case we fail?

Reply:
why, we would jam the whole exit queue just because of a single funny asset?